src/runtime/runtime.ts

```diff
@@ -177,8 +206,20 @@ export class Runtime {
 
   // stuff the strategizer needs
 
-}
+  // stuff from shells/lib/utils
 
-let runtime: Runtime = null;
+  // TODO(sjmiles): there is redundancy vs `parse/loadManifest` above, but this is
+  // temporary until we polish the Utils integration.
 
+  async parse(content: string, options?): Promise<Manifest> {
+    const {loader} = this;
+    const id = `in-memory-${Math.floor((Math.random()+1)*1e6)}.manifest`;
```

Comment:
This probably also warrants a TODO or GitHub issue to have design thought put into in-memory manifest id generation to avoid conflicts. @Cypher1

Reply:
Note that Utils has been doing this thing for a long time. To be clear, I'm happy to follow runtime-team's suggestions/design-rules, I'm only noting this isn't actually _new_.

Reply:
Thanks @shaper and @sjmiles I think there's a bit refactoring to be done to the parser, hopefully it accomodates this better.